tiny_cnn/layers/average_pooling_layer.h

```diff
@@ -59,15 +59,44 @@ class average_pooling_layer : public partial_connected_layer<Activation> {
             : Base(in_width * in_height * in_channels,
                    in_width * in_height * in_channels / sqr(pooling_size),
                    in_channels, in_channels, float_t(1) / sqr(pooling_size)),
-              stride_(pooling_size),
+              stride_width_(pooling_size),
+              stride_height_(pooling_size),
               in_(in_width, in_height, in_channels),
               out_(in_width/pooling_size, in_height/pooling_size, in_channels),
               w_(pooling_size, pooling_size, in_channels) {
         if ((in_width % pooling_size) || (in_height % pooling_size)) {
             pooling_size_mismatch(in_width, in_height, pooling_size);
         }
 
-        init_connection(pooling_size);
+        init_connection(pooling_size, pooling_size);
+    }
+
+    /**
+     * @param in_width     [in] width of input image
+     * @param in_height    [in] height of input image
+     * @param in_channels  [in] the number of input image channels(depth)
+     * @param pooling_size [in] factor by which to downscale
+     **/
+    average_pooling_layer(cnn_size_t in_width,
+                          cnn_size_t in_height,
+                          cnn_size_t in_channels,
+                          std::pair<cnn_size_t, cnn_size_t> pooling_size)
```

Comment:
@edgarriba if I use two `cnn_size_t` the ctor collides with the already existing ones.